### PR TITLE
Recreation and optimizing 

### PR DESCRIPTION
**Pull request by synecdochenoire**
Make MapVC more readable and flexible for changes also now its more safety

### CODE REVIEW

File: Placeeo/Controller/MainViewController.swift [hunk 1]
```diff
@@ -6,22 +6,30 @@
 //
 
 import UIKit
+import CoreLocation
+import MapKit
 
-class ViewController: UIViewController {
+class MainViewController: MapViewController{
```
Comment (isidar) [inline Placeeo/Controller/MainViewController.swift]: why did you change `UIViewController` to `MapViewController`?

**Review by isidar — CHANGES_REQUESTED** [review]
too many instances of MapViewController and none of them are used. try using one

File: Placeeo/Controller/MainViewController.swift [hunk 2]
```diff
@@ -6,22 +6,30 @@
...
-        let mapViewController = MapViewController()
         
-        mapViewController.show(places: placesArray)
-       
-        present(mapViewController, animated: true, completion: nil)
+          let placeOne = Place(name: "Castle Uzhgorod", lat: 48.62216095313416, long: 22.308029690972464)
+          let placeTwo = Place(name: "Church", lat: 48.623380761219515, long: 22.302193204377218)
+          
+          
+          let places = [placeOne, placeTwo]
```
Comment (isidar) [inline Placeeo/Controller/MainViewController.swift]: where do you use these places?

Comment (synecdochenoire) [reply Placeeo/Controller/MainViewController.swift]: its for future using and creating MKAnottation

File: Placeeo/Controller/MainViewController.swift (hunk 1 above)
Comment (synecdochenoire) [reply Placeeo/Controller/MainViewController.swift]: i think than in future i need to inherit from MapVC

File: Placeeo/Controller/MainViewController.swift [hunk 3]
```diff
@@ -6,22 +6,30 @@
...
+import CoreLocation
+import MapKit
 
-class ViewController: UIViewController {
+class MainViewController: MapViewController{
 
-    var placesArray: [Place] = []
-  
-    
+  let mapVC = MapViewController()
```
Comment (isidar) [inline Placeeo/Controller/MainViewController.swift]: it's not used anywhere

File: Placeeo/Controller/MainViewController.swift [hunk 4]
```diff
@@ -6,22 +6,30 @@
...
+          let placeOne = Place(name: "Castle Uzhgorod", lat: 48.62216095313416, long: 22.308029690972464)
+          let placeTwo = Place(name: "Church", lat: 48.623380761219515, long: 22.302193204377218)
+          
+          
+          let places = [placeOne, placeTwo]
+          
+   
+        
+        
+      let mapViewController = MapViewController()
```
Comment (isidar) [inline Placeeo/Controller/MainViewController.swift]: it's not used anywhere

**Review by isidar — CHANGES_REQUESTED** [review]

File: Placeeo/Controller/MainViewController.swift (hunk 1 above)
Comment (isidar) [reply Placeeo/Controller/MainViewController.swift]: not really. Main and Map view controllers must not to depend on each other, thus none of them should inherit each other. let's create an instance of MapVC inside MainVC and call it's methods to show places

File: Placeeo/Controller/MainViewController.swift (hunk 2 above)
Comment (isidar) [reply Placeeo/Controller/MainViewController.swift]: I see. it's the right place for them, but let's not commit the things which are not used anywhere. you can have them locally and commit them later if needed

File: Placeeo/Controller/MainViewController.swift (hunk 1 above)
Comment (synecdochenoire) [reply Placeeo/Controller/MainViewController.swift]: done!

File: Placeeo/Controller/MainViewController.swift (hunk 2 above)
Comment (synecdochenoire) [reply Placeeo/Controller/MainViewController.swift]: check again!

File: Placeeo/Controller/MainViewController.swift (hunk 3 above)
Comment (synecdochenoire) [reply Placeeo/Controller/MainViewController.swift]: deleted

File: Placeeo/Controller/MainViewController.swift (hunk 4 above)
Comment (synecdochenoire) [reply Placeeo/Controller/MainViewController.swift]: deleted

**Comment by isidar** [comment]
good job!

**Review by isidar — APPROVED** [review]
